### DISCUSSION TITLE
Keycloak doesn't load custom providers

### DISCUSSION THREAD

**Discussion by s-apanasionak** [keycloak/keycloak#11713] [Archive]
I use IDELauncher to start keycloak from IDE.
I use custom UserStorageProvider (jar file). I put it in directory /home/svyatoslav/IdeaProjects/keycloak/home/providers
And specify next vm option -Dkc.home.dir=/home/svyatoslav/IdeaProjects/keycloak/home
After keycloak started I do not see my custom provider in list on user federation tab.
How can I make keycloak to see my custom provider?

**Comment by eggyknap** [comment]
I haven't tried running keycloak in an IDE, but with the providers I've been writing, I've needed to run `kc.sh build` after putting the .jar in the `providers` directory, before Keycloak will notice them.

**Reply by s-apanasionak** [reply]
I have no problems with providers when keycloak is running in not IDE.
Eventually, I used remote debugging.
But I still don't know how to make keycloak use providers when running from IDE.
